src/core/unit.c

```diff
@@ -1929,6 +1929,102 @@ static bool unit_verify_deps(Unit *u) {
         return true;
 }
 
+static bool unit_can_instantiate(Unit *u, char **ret) {
+        _cleanup_free_ char *t = NULL;
+        unsigned n;
+        int r;
+
+        assert(u);
+        assert(u->manager);
+
+        if (!u->instance)
+                return true;
+
+        if (u->n_max_instances == 0)
```

Comment:
I think n_max_instances == 0 should mean "do not allow this unit to be started". 

and UINT_MAX should mean "unbounded".

Reply:
in fact I think it would make sense to extend the concept to non-instance units, so that people can set the value to 0 of those, so that the unit can never be started.